roles/ceph-osd/tasks/check_devices.yml

```diff
@@ -5,7 +5,7 @@
 # As as a final word, I prefer to keep the partition check instead of running ceph-disk prepare with "failed_when: false"
 # I believe it's safer
 - name: check if the device is a partition or a disk
-  shell: "echo '{{ item }}' | egrep '/dev/(([a-z]{3,4}[0-9]$)|(cciss/c[0-9]{1}d[0-9]{1}p[0-9]$))'"
+  shell: "echo '{{ item }}' | egrep '/dev/[sd[a-z]{1,2}|hd[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p][0-9]{1,2}$'"
```

Comment:
Sebastien, you were right, the regex I specified is wrong, thanks for pointing it out.  Here's a correction to the regex syntax along with a quick method of testing it.

# echo /dev/hdab99 | egrep '/dev/(sd[a-z]{1,2}|hd[a-z]{1,2}|cciss/c[0-9]d[0-9]{1,2}p|nvme[0-9]n[0-9]p)[0-9]{1,2}$' ; echo $?
/dev/hdab99
0

So I think this works, try running different device names through it and see if it works for you.  Again, the first part of the regex represents the different flavors of device names, and the last part is the partition number syntax.  Does this work for you?  I tried to limit to 99 disks/server, since Ceph is designed for scale-out, not to cram as many disks as possible into 1 server.    But I wanted to have option to have >= 10 partitions per device, since in the case of NVM SSDs we may need to get away with that sometimes.

If you agree with it I'll submit a new pull request with just this change.

-ben